rest_witchcraft/filters.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+"""
+Provides generic filtering backends that can be used to filter the results
+returned by list views.
+"""
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+from sqlalchemy import func, or_
+from sqlalchemy.sql import operators
+
+from django.template import loader
+from django.utils.encoding import force_text
+from django.utils.translation import gettext_lazy
+
+from rest_framework.compat import coreapi, coreschema
+from rest_framework.filters import BaseFilterBackend
+from rest_framework.settings import api_settings
+
+
+class SearchFilter(BaseFilterBackend):
```

Comment:
not happy with url filter?

Reply:
just matching drf builtins, see https://github.com/encode/django-rest-framework/blob/master/rest_framework/filters.py